server/controllers/GroupController.js

```diff
@@ -0,0 +1,125 @@
+import db from '../models/dbQuery';
+
+const Group = {
+  /**
+   * create user group
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} group
+   */
+  async createGroup(req, res) {
+    if (!req.body.group_name || !req.body.role) {
+      return res.status(400).send({ message: 'All fields are required' });
+    }
+    const createQuery = `INSERT INTO groups (name, role, owner_id) VALUES ($1, $2, $3) returning *`;
+
+    const values = [
+      req.body.group_name,
+      req.body.role,
+      req.user.id
+    ];
+
+    try {
+      const { rows } = await db.query(createQuery, values);
+      return res.status(201).json({
+        success: true,
+        message: 'Group Created successfully',
+        data: rows[0]
+      });
+    } catch (error) {
+      return res.status(400).json({ error: error });
+    }
+  },
+
+  /**
+   * Get all groups
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} all created group
+   */
+
+  async allGroups(req, res) {
+
+    const getQuery = `SELECT * FROM groups`;
+
+    try {
+      const { rows, rowCount } = await db.query(getQuery);
+      return res.status(200).json({
+        success: 'true',
+        message: 'All created groups retrieved successfully',
+        data: rows,
+        count: rowCount
+      });
+    } catch (error) {
+      return res.status(400).json({ error: error });
+    }
+  },
+
+  /**
+   * Add user to a created group
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} object
+   */
+  async addUser(req, res) {
+    if (!req.body.group_name || !req.body.role) {
```

Comment:
Nice work. You need to check the db to know if the `group` exists before running the insert query